### PR TITLE
add web-animations types

### PR DESCRIPTION
**Pull request by benbraou**
Added web animation types based on [this](https://developer.mozilla.org/en-US/docs/Web/API/Animation/Animation)

Fixes Microsoft/TypeScript#21492

### CODE REVIEW

File: baselines/dom.generated.d.ts [hunk 1]
```diff
@@ -14852,6 +14853,59 @@ interface EventSourceInit {
     readonly withCredentials: boolean;
 }
 
+interface AnimationOptions {
+    id?: DOMString;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: we so not use this type else where. just use `string`.

File: baselines/dom.generated.d.ts [hunk 2]
```diff
@@ -14852,6 +14853,59 @@ interface EventSourceInit {
...
+    delay?: number;
+    direction?: "normal" | "reverse" | "alternate" | "alternate-reverse";
+    duration?: number;
+    easing?: string;
+    endDelay?: number;
+    fill?: "none" | "forwards" | "backwards" | "both"| "auto";
+    iterationStart?: number;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: missing `iterations: number`

File: baselines/dom.generated.d.ts [hunk 3]
```diff
@@ -15360,6 +15414,8 @@ type ScrollRestoration = "auto" | "manual";
 type FormDataEntryValue = string | File;
 type InsertPosition = "beforebegin" | "afterbegin" | "beforeend" | "afterend";
 type HeadersInit = Headers | string[][] | { [key: string]: string };
+type DOMString = string;
+type AnimationKeyFrame = {[key: string]: string | number | [string | number, string | number] | undefined};
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: the spec does not say it has to be a tuple.. there are examples https://drafts.csswg.org/web-animations/#example-be6aa998 where it is an array of strings.

Comment (mhegazy) [reply baselines/dom.generated.d.ts]: `null` seems to be allowed as well

File: baselines/dom.generated.d.ts [hunk 4]
```diff
@@ -14852,6 +14853,59 @@ interface EventSourceInit {
...
+    readonly finished: Promise<Animation>;
+    id: string;
+    readonly pending: boolean;
+    readonly playState: "idle" | "running" | "paused" | "finished";
+    playbackRate: number;
+    readonly ready: Promise<Animation>;
+    startTime: number;
+    timeline: AnimationTimeline;
+    oncancel() => void;
+    oncancel() => void;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: repeated.

File: baselines/dom.generated.d.ts [hunk 5]
```diff
@@ -14852,6 +14853,59 @@ interface EventSourceInit {
...
+    effect: AnimationEffectReadOnly;
+    readonly finished: Promise<Animation>;
+    id: string;
+    readonly pending: boolean;
+    readonly playState: "idle" | "running" | "paused" | "finished";
+    playbackRate: number;
+    readonly ready: Promise<Animation>;
+    startTime: number;
+    timeline: AnimationTimeline;
+    oncancel() => void;
```
Comment (mhegazy) [inline baselines/dom.generated.d.ts]: oncancel(): void;

Comment (mhegazy) [reply baselines/dom.generated.d.ts]: missing `onFinish`

File: baselines/dom.generated.d.ts (hunk 1 above)
Comment (benbraou) [reply baselines/dom.generated.d.ts]: ok => commit will be amended accordingly

File: baselines/dom.generated.d.ts (hunk 2 above)
Comment (benbraou) [reply baselines/dom.generated.d.ts]: ok => commit will be amended accordingly

File: baselines/dom.generated.d.ts (hunk 4 above)
Comment (benbraou) [reply baselines/dom.generated.d.ts]: ok => commit will be amended accordingly

File: baselines/dom.generated.d.ts (hunk 5 above)
Comment (benbraou) [reply baselines/dom.generated.d.ts]: To be more specific about the event type
`oncancel: (this: Animation, ev: AnimationPlaybackEvent) => any;`
I will add the interface `AnimationPlaybackEvent `

File: baselines/dom.generated.d.ts (hunk 3 above)
Comment (benbraou) [reply baselines/dom.generated.d.ts]: Done: I defined the type
```typescript
{offset?: number | null | (number | null)[]} & {[key: string]: string | number | number[] | string[]};
```

**Comment by benbraou** [comment]
@mhegazy I have updated the PR to address your review (I added on extra commit, please let me know if I need to squash)
Thanks

**Pull request merged**